elytron/src/main/java/org/wildfly/extension/elytron/ElytronPersistentResourceXMLDescriptionFactory.java

```diff
@@ -0,0 +1,228 @@
+import static org.wildfly.extension.elytron.ElytronDescriptionConstants.EXPRESSION;
+import static org.wildfly.extension.elytron.ElytronDescriptionConstants.EXPRESSION_RESOLVER;
+import static org.wildfly.extension.elytron.ElytronDescriptionConstants.PERMISSION_SETS;
+import static org.wildfly.extension.elytron.ElytronDescriptionConstants.SECURITY_PROPERTY;
+import static org.wildfly.extension.elytron.PermissionMapperDefinitions.PERMISSIONS;
+
+public enum ElytronPersistentResourceXMLDescriptionFactory implements Function<ElytronSubsystemSchema, PersistentResourceXMLDescription> {
+    INSTANCE;
+
+    @Override
+    public PersistentResourceXMLDescription apply(ElytronSubsystemSchema schema) {
+        PersistentResourceXMLDescription.PersistentResourceXMLBuilder builder = builder(ElytronExtension.SUBSYSTEM_PATH, schema.getNamespace());
+        if (schema.since(ElytronSubsystemSchema.VERSION_1_0)) {
+            builder.addAttribute(ElytronDefinition.DEFAULT_AUTHENTICATION_CONTEXT)
+                    .addAttribute(ElytronDefinition.INITIAL_PROVIDERS)
+                    .addAttribute(ElytronDefinition.FINAL_PROVIDERS)
+                    .addAttribute(ElytronDefinition.DISALLOWED_PROVIDERS)
+                    .addAttribute(ElytronDefinition.SECURITY_PROPERTIES, new AttributeParsers.PropertiesParser(null, SECURITY_PROPERTY, true), new AttributeMarshallers.PropertiesAttributeMarshaller(null, SECURITY_PROPERTY, true));
+        }
+
+        if (schema.since(ElytronSubsystemSchema.VERSION_5_0)) {
+            builder.addAttribute(ElytronDefinition.REGISTER_JASPI_FACTORY)
+                    .addAttribute(ElytronDefinition.DEFAULT_SSL_CONTEXT);
+        }
+
+        addAuthenticationClientParser(builder, schema);
+
+        if (schema.since(ElytronSubsystemSchema.VERSION_1_0)) {
+            builder.addChild(new ProviderParser().parser);
+        }
+
+        addAuditLoginParser(builder, schema);
+        addSecurityDomainParser(schema, builder);
+        addRealmParser(schema, builder);
+
+        if (schema.since(ElytronSubsystemSchema.VERSION_1_0)) {
+            builder.addChild(new CredentialSecurityFactoryParser().parser);
+        }
+
+        addMapperParser(schema, builder);
+
+        if (schema.since(ElytronSubsystemSchema.VERSION_3_0)) {
+            builder.addChild(getPermissionSetParser());
+        }
+
+        if (schema.since(ElytronSubsystemSchema.VERSION_1_0)) {
+            builder.addChild(new HttpParser().parser);
+        }
+
+        if (schema.since(ElytronSubsystemSchema.VERSION_1_0)) {
+            builder.addChild(new SaslParser().parser);
+        }
+
+        addTlsParser(schema, builder);
+        addCredentialStoreParser(schema, builder);
+
+        if (schema.since(ElytronSubsystemSchema.VERSION_13_0)) {
+            builder.addChild(getExpressionResolverParser());
+        }
+
+        if (schema.since(ElytronSubsystemSchema.VERSION_3_0)) {
```

Comment:
Should this be schema.since(ElytronSubsystemSchema.VERSION_1_0)?

Reply:
Fixed